modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/service/DLOpenerOneDriveDLAppServiceWrapper.java

```diff
@@ -162,6 +163,21 @@ public void checkInFileEntry(
 			serviceContext.getUserId(), fileEntry);
 	}
 
+	@Override
+	public FileEntry updateFileEntryAndCheckIn(
+			long fileEntryId, String sourceFileName, String mimeType,
+			String title, String description, String changeLog,
+			DLVersionNumberIncrease dlVersionNumberIncrease,
+			InputStream inputStream, long size, ServiceContext serviceContext)
+		throws PortalException {
+
+		FileEntry fileEntry = getFileEntry(fileEntryId);
```

Comment:
Shouldn't this be:
```
		checkInFileEntry(
			fileEntryId, dlVersionNumberIncrease, changeLog, serviceContext);

		return getFileEntry(fileEntryId);
```

to ensure we return a fresh FileEntry object?

Reply:
Also, it may be worth to delegate if OneDrive is not enabled:
```
		if (!_dlOpenerOneDriveManager.isConfigured(fileEntry.getCompanyId()) ||
			!_dlOpenerOneDriveManager.isOneDriveFile(fileEntry)) {

			return super.updateFileEntryAndCheckIn(...);
		}
```